src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -213,6 +240,66 @@ public long getCurrentSpoolerSize() {
     public SpoolerConfig getSpoolConfig() {
         return config;
     }
-}
 
+    /**
+     * Extract message ids from the persistenceSpool plugin's on disk database and insert the message \
+     * ids into queueOfMessageId, this function is only used in FileSystem storage mode.
+     * @param diskQueueOfIds list of messageIds to sync
+     * @param persistenceSpool instance of CloudMessageSpool
+     */
+    private void persistentQueueSync(List<Long> diskQueueOfIds, CloudMessageSpool persistenceSpool)
+            throws InterruptedException, ServiceLoadException {
+        if (!diskQueueOfIds.isEmpty()) {
+            long highestId = -1;
+            int i;
+            for (i = 0; i < diskQueueOfIds.size(); i += 1) {
```

Comment:
use an enhanced (for-each) for loop.

Reply:
Okay I will implement an enhanced for loop